### PR TITLE
Add SegFormer Sugar Beets 2016 training and diagnostics tutorial

### PR DESCRIPTION
**Pull request by everley3lc**
# Add SegFormer Sugar Beets 2016 training + diagnostics tutorial

## Summary
Adds a semantic-segmentation tutorial that fine-tunes **SegFormer** on the University of Bonn
**Sugar Beets 2016** dataset, then uses 3LC to collect per-sample metrics and demonstrate two
data-curation workflows.

## What's included
- `tutorials/3-training-and-metrics/huggingface-sugarbeets-segformer-training.ipynb` — one notebook, two parts.
- `tutorials/images/sugarbeets/` — 5 figures (overview + Part 2 screenshots).
- One new entry in `tutorials/notebooks.yaml`.

## The tutorial
A 3-class semantic-segmentation problem — **soil / sugarbeet (crop) / weed** — in two parts.

**Part 1 — Training & diagnostics (runnable):**
- Builds two 3LC `Table`s (train / val, seeded 80/20 split) with RLE-backed segmentation masks and an
  editable per-row sample weight.
- Fine-tunes SegFormer (HuggingFace `transformers`, `nvidia/mit-b0` encoder) with a per-class ×
  per-sample weighted cross-entropy and differential encoder/head learning rates.
- **Training run:** logs per-epoch validation mIoU + per-class IoU, checkpoints the best model, and on
  collection epochs writes per-sample metrics (predicted mask, IoU, loss, entropy, pooled embedding).
- **Diagnostics run:** reloads the best model and collects richer per-sample diagnostics in one shared
  embedding space — predicted mask, error map, confidence map, per-class IoU, and per-class GT-masked
  and prediction-masked embeddings.

**Part 2 — Data curation (narrative + screenshots):**
- **Data reduction** — use the per-class traversal index to prune redundant, low-information samples.
- **Correcting annotations** — surface likely-mislabeled ground truth via the error map's mispredicted
  pixel count and low per-class IoU, then adopt the model's prediction where the ground truth is wrong.

## Model & data
- **Model:** SegFormer `nvidia/mit-b0` (ImageNet-pretrained) via `transformers`.
- **Data:** University of Bonn Sugar Beets 2016 (Chebrolu et al., IJRR 2017), prepared into a flat
  `images/` · `masks/` · `splits/` layout (single-channel class-indexed masks: `0=soil, 1=sugarbeet,
  2=weed`; deterministic 80/20 split). The notebook includes full instructions to prepare the dataset
  from the raw source; running locally is a one-line `DOWNLOAD_PATH` change.

## 3LC features demonstrated
Semantic-segmentation tables, per-sample metrics collection, pooled and per-class masked embeddings
with shared 2D reduction, a separate diagnostics run on the best model, error/confidence maps,
per-class IoU, editable sample weights, and Dashboard-driven curation (traversal-index reduction and
annotation correction).

### CODE REVIEW

File: tutorials/notebooks.yaml [hunk 1]
```diff
@@ -819,6 +819,33 @@
     EPOCHS: {test: 1, publish: 10}
   depends_on: [1-create-tables/semantic-segmentation/create-pascal-voc-semseg-table.ipynb]
   creates: []
+- path: 3-training-and-metrics/huggingface-sugarbeets-segformer-training.ipynb
+  title: Train and Diagnose SegFormer on Sugar Beets 2016
+  blurb: In this tutorial, we fine-tune a SegFormer semantic-segmentation model on the University of Bonn Sugar Beets 2016 agricultural dataset. This is a 3-class problem — soil, sugarbeet (crop), and weed....
```
Comment (KellyWalker) [inline tutorials/notebooks.yaml]: Did you mean to end with sentence with ....?

Comment (everley3lc) [reply tutorials/notebooks.yaml]: Fixed.

**Comment by everley3lc** [comment]
Comment was updated, "Testing" section removed.

**Review by KellyWalker — APPROVED** [review]

**Review by KellyWalker — APPROVED** [review]
